### PR TITLE
Entities configuration

### PR DESCRIPTION
**Pull request by caroo7**
Removed unnecessary dependencies.
Use project lombok if necessary.
Added entities according to database schema and spring data repositories for them.
Added integration tests for repositories.

### CODE REVIEW

File: database/src/main/java/entities/Book.java [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
 import java.io.Serializable;
+import java.sql.Timestamp;
+import java.util.Set;
 
+// not sure about model classes serialization, maybe later we create some kind of DTO classes to passing over HTTP
```
Comment (LIttleAncientForestKami) [inline database/src/main/java/entities/Book.java:12]: :+1:

File: database/src/main/java/entities/Book.java [hunk 2]
```diff
@@ -17,17 +24,29 @@
...
 
+    @Column(length = 10000)
     private String description;
 
-    // add library list (entity), tag list (entity), category id (entity), discount type (enum)
+    private String discount;
+
+    private String price;
+
+    private Timestamp timestamp;
```
Comment (LIttleAncientForestKami) [inline database/src/main/java/entities/Book.java:34]: Careful with Timestamp. Remember to annotate it properly, depending on time type you wanna use on DB side.

**Comment by LIttleAncientForestKami** [comment]
:ok:

File: database/src/main/java/entities/Genre.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import java.io.Serializable;
+
+@Entity
+@NoArgsConstructor
+public class Genre implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
```
Comment (egolesor) [inline database/src/main/java/entities/Genre.java:18]: I want to know if the field is private and there is no getter, Is hibernate able to read the state of the genre object ?

File: database/src/main/java/entities/Tag.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import java.io.Serializable;
+
+@Entity
+@NoArgsConstructor
+public class Tag implements Serializable {
```
Comment (egolesor) [inline database/src/main/java/entities/Tag.java:15]: Why its better to store the tags on such objects not in enum ?

**Comment by egolesor** [comment]
:ok_hand:

**Pull request merged**